src/go/rpk/pkg/config/params_test.go

```diff
@@ -1204,3 +1204,162 @@ func TestXSetDefaultsPaths(t *testing.T) {
 		}
 	}
 }
+
+func TestConfig_fixSchemePorts(t *testing.T) {
+	tests := []struct {
+		name   string
+		config Config
+		expect Config
+		errMsg string
+	}{
+		{
+			name: "fix missing ports in redpanda.yaml addresses",
+			config: Config{
+				redpandaYaml: RedpandaYaml{
+					Rpk: RpkNodeConfig{
+						KafkaAPI: RpkKafkaAPI{
+							Brokers: []string{"broker1", "broker2:9093"},
+						},
+						AdminAPI: RpkAdminAPI{
+							Addresses: []string{"address1", "address2:2222"},
+						},
+					},
+				},
+			},
+			expect: Config{
+				redpandaYaml: RedpandaYaml{
+					Rpk: RpkNodeConfig{
+						KafkaAPI: RpkKafkaAPI{
+							Brokers: []string{"broker1:9092", "broker2:9093"},
+						},
+						AdminAPI: RpkAdminAPI{
+							Addresses: []string{"address1:9644", "address2:2222"},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "fix missing ports in profile brokers",
+			config: Config{
+				rpkYaml: RpkYaml{
+					CurrentProfile: "default",
+					Profiles: []RpkProfile{
+						{
+							Name: "default",
+							KafkaAPI: RpkKafkaAPI{
+								Brokers: []string{"profile-broker1", "profile-broker2:9094"},
+							},
+							AdminAPI: RpkAdminAPI{
+								Addresses: []string{"address1", "address2:2222"},
+							},
+							SR: RpkSchemaRegistryAPI{
+								Addresses: []string{"address1", "address2", "address3:8088"},
+							},
+						},
+					},
+				},
+			},
+			expect: Config{
+				rpkYaml: RpkYaml{
+					CurrentProfile: "default",
+					Profiles: []RpkProfile{
+						{
+							Name: "default",
+							KafkaAPI: RpkKafkaAPI{
+								Brokers: []string{"profile-broker1:9092", "profile-broker2:9094"},
+							},
+							AdminAPI: RpkAdminAPI{
+								Addresses: []string{"address1:9644", "address2:2222"},
+							},
+							SR: RpkSchemaRegistryAPI{
+								Addresses: []string{"address1:8081", "address2:8081", "address3:8088"},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "fix missing ports in redpanda.yaml addresses",
+			config: Config{
+				redpandaYaml: RedpandaYaml{
+					Rpk: RpkNodeConfig{
+						KafkaAPI: RpkKafkaAPI{
+							Brokers: []string{"broker1", "broker2:9093"},
+						},
+						AdminAPI: RpkAdminAPI{
+							Addresses: []string{"address1", "address2:2222"},
+						},
+					},
+				},
+			},
+			expect: Config{
+				redpandaYaml: RedpandaYaml{
+					Rpk: RpkNodeConfig{
+						KafkaAPI: RpkKafkaAPI{
+							Brokers: []string{"broker1:9092", "broker2:9093"},
+						},
+						AdminAPI: RpkAdminAPI{
+							Addresses: []string{"address1:9644", "address2:2222"},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "fix missing ports in redpanda.yaml addresses with IPv6",
```

Comment:
Maybe worth to add another test for Ipv4 too?

Reply:
Yes, good point. I'll add that to the main branch then to avoid drifting this commit from the original (since this is a cherry-pick)